packages/components/button/src/__tests__/Button.spec.tsx

```diff
@@ -1,15 +1,192 @@
-import { render, screen } from '@testing-library/react';
+import { render, screen, fireEvent } from '@testing-library/react';
 import Button from '../Button';
 import { ThemeProvider } from '@douro-ui/react';
+import { ButtonProps } from '../button.types';
 
-describe('<Button />', () => {
-  it('should render properly', () => {
-    render(
-      <ThemeProvider>
-        <Button label="Foo" />
-      </ThemeProvider>,
+const renderTheBtnType = (
+  label: string,
+  typeBtn: ButtonProps['typeBtn'],
+  disabled: boolean,
+  size?: ButtonProps['size'],
+  onClick?: () => void,
+) => {
+  return render(
+    <ThemeProvider>
+      <Button
+        label={label}
+        typeBtn={typeBtn}
+        disabled={disabled}
+        size={size}
+        onClick={onClick}
+      />
+    </ThemeProvider>,
+  );
+};
+
+describe('<Button >', () => {
+  it('should render a clickable primary button', () => {
+    renderTheBtnType('Primary button and available', 'primary', false);
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Primary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
```

Comment:
The expect doesn't match with the message of the text. If you are just testing if the component is rendered, do not use a random prop like disable just to expect something.